src/nested_pandas/nestedframe/core.py

```diff
@@ -0,0 +1,61 @@
+# typing.Self and "|" union syntax don't exist in Python 3.9
+from __future__ import annotations
+
+import pandas as pd
+
+from nested_pandas.series import packer
+
+
+class NestedFrame(pd.DataFrame):
+    """A Pandas Dataframe extension with support for nested structure.
+
+    See https://pandas.pydata.org/docs/development/extending.html#subclassing-pandas-data-structures
+    """
+
+    # normal properties
+    _metadata = ["added_property"]
+
+    @property
+    def _constructor(self) -> Self:  # type: ignore[name-defined] # noqa: F821
+        return NestedFrame
+
+    @property
+    def _constructor_expanddim(self) -> Self:  # type: ignore[name-defined] # noqa: F821
+        return NestedFrame
+
+    @property
+    def all_columns(self) -> dict:
+        """returns a dictionary of columns for each base/nested dataframe"""
+        all_columns = {"base": self.columns}
+        for column in self.columns:
+            if hasattr(self[column], "nest"):
```

Comment:
I would check column's dtype instead, it should be `NestedDtype`. No code suggestion here, because we need to import it first
```python
from nested_pandas import NestedDtype

if isinstance(self[column].dtype, NestedDtype):
```